MCP: fix spawn ... ENOENT issue

shemhamforash23 opened the pull request:
## Description
This PR fixes an issue with unavailable shell commands due to incorrect PATH variable propagation. 
It is now being propagated to each spawned MCP server process in transport. 

It also adds MCP server log output for debugging purposes.

### Related issues:
https://github.com/raycast/extensions/issues/18080
https://github.com/raycast/extensions/issues/17961
https://github.com/raycast/extensions/issues/17843

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

EvanZhouDev reviewed (changes requested):
Code changes looks good! I realize the `MCP` fix is likely on Raycast linting, but it's still more conventional in this case to write `MCP` instead of `Mcp`.

strikerlulu commented on extensions/mcp/src/getProcessedClients.ts:
```diff
@@ -1,49 +1,83 @@
+  // Get environment variables from shell
+  try {
+    // Get PATH from shell
+    const pathCommand = `${shell} -l -i -c 'echo $PATH'`;
```
can we also remove interactive `-i` flag ?
the command execution never gets completed in my shell with this flag

strikerlulu replied:
also in src/manage-mcp-servers.tsx#98

shemhamforash23 replied:
Okay, removed `-i` flags from both

shemhamforash23 commented on extensions/mcp/package.json:
```diff
@@ -16,20 +22,20 @@
     {
       "name": "manage-mcp-servers",
-      "title": "Manage MCP Servers",
+      "title": "Manage Mcp Servers",
```
yes, this change was made because of linter errors. Is there a way to ignore linter for this case?

shemhamforash23 commented:
Hey!

Made changes according to review. 

Also added a couple of new features:
- added timeout option for long-running tools (for example, deep research may take up to 10 minutes to complete). 60 seconds by default.
- getting list of tools now works asynchronously, drastically increasing performance of tools list call.
- tools that are failed to start are now propagated to the response with an error, that caused them to fail. 
- fixed failure when parsing empty string as JSON tool arguments.

pernielsentikaer commented:
@EvanZhouDev can you check this again?

EvanZhouDev commented:
Code changes look good to me!
